### PR TITLE
extended container testcases

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py [hunk 1]
```diff
@@ -153,9 +154,10 @@ def get_containers_containerid_processes(self, nodeid, containername):
         finally:
             return response
 
-    def post_containers_containerid_processes(self, nodeid, containername, data):
+    def post_containers_containerid_jobs(self, nodeid, containername, data):
+        import ipdb;ipdb.set_trace()
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py]: Please don't push ipdb code

File: functional_testing/Grid_API_Testing/api_testing/python_client/client.py [hunk 2]
```diff
@@ -203,3 +203,51 @@ def wait_on_container_job_update(self, container_name, job_id, timeout, removed)
...
+        containers = self.client.container.find('ovs')
+        ovs_exist = [key for key, value in containers.items()]
+        if not ovs_exist:
+            ovs_flist = "https://hub.gig.tech/gig-official-apps/ovs.flist"
+            ovs = int(self.client.container.create(ovs_flist, host_network=True , tags=['ovs']).get().data)
+            self.ovs_client = self.client.container.client(ovs)
+            time.sleep(2)
+            self.ovs_client.json('ovs.bridge-add', {"bridge": "backplane"})
+            self.ovs_client.json('ovs.vlan-ensure', {'master': 'backplane', 'vlan': 2000, 'name': 'vxbackend'})
+        else:
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/python_client/client.py]: why should we have else statement here?

Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/python_client/client.py]: yes , cause we can't create more than one ovs container in node

File: functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py (hunk 1 above)
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py]: done , sorry

File: functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py [hunk 3]
```diff
@@ -206,3 +213,722 @@ def test005_Check_container_access_to_host_dev(self):
...
+        self.assertEqual(response.state, "ERROR")
+
+        self.lg.info("Check that C1  can ping second container C2 through its gateway, should succeed.")
+        response = C1_client.bash("ping -w2 %s "%gateway).get()
+        self.assertEqual(response.state, "SUCCESS")
+
+    def test012_create_container_with_dns_in_config(self):
+        """ GAT-093
+
+        *Test case for test creation of containers with different network and with dns *
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py]: wrong description

File: functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py [hunk 4]
```diff
@@ -206,3 +213,722 @@ def test005_Check_container_access_to_host_dev(self):
...
+        C2_client = self.g8core.get_container_client(C2_name)
+        response = C2_client.bash('cat /etc/resolv.conf | grep %s'%dns).get()
+        self.assertEqual(response.state, "SUCCESS")
+        response = C2_client.bash('ping -c 2 %s'%dns).get()
+        self.assertEqual(response.state, "SUCCESS")
+
+    def test013_create_container_with_filesystem(self):
+        """ GAT-094
+
+        *Test case for test creation of containers with different network and with dns *
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py]: wrong description

File: functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py [hunk 5]
```diff
@@ -206,3 +213,722 @@ def test005_Check_container_access_to_host_dev(self):
...
+
+        self.lg.info("Check that created file doesn't exicst in C2.")
+
+        response = C1_client.bash("ls | grep %s"%file_name).get()
+        self.assertEqual(response.state, 'SUCCESS')
+
+        response = C2_client.bash("ls | grep %s"%file_name).get()
+        self.assertEqual(response.state, 'ERROR')
+
+    def test015_Writing_in_containers_files_create_containers_with_open_ports(self):
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py]: wrong name

File: functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py [hunk 6]
```diff
@@ -206,3 +213,722 @@ def test005_Check_container_access_to_host_dev(self):
...
+        C3_client = self.g8core.get_container_client(C3_name)
+
+        self.lg.info("Check if third container (c3) can ping first container (c1), should fail.")
+        response = C3_client.bash('ping -c 10 %s'%C1_br_ip).get()
+        self.assertEqual(response.state, 'ERROR')
+
+        self.lg.info("Delete created bridge ")
+        self.bridges_api.delete_nodes_bridges_bridgeid(self.node_id, bridge_name)
+
+    def test007_create_containers_with_multi_bridge(self):
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py [hunk 7]
```diff
@@ -206,3 +213,722 @@ def test005_Check_container_access_to_host_dev(self):
...
+        """ GAT-088
+        *Test case for create containers with same bridge and make sure they can connect to each other *
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py [hunk 8]
```diff
@@ -206,3 +213,722 @@ def test005_Check_container_access_to_host_dev(self):
...
+        response = C2_client.bash('ping -w 5 %s'%C1_br_ip).get()
+        self.assertEqual(response.state, 'ERROR')
+
+        self.lg.info("Delete created bridge ")
+        self.bridges_api.delete_nodes_bridges_bridgeid(self.node_id, B2_name)
+        self.bridges_api.delete_nodes_bridges_bridgeid(self.node_id, B1_name)
+
+    def test008_Create_container_with_zerotier_network(self):
+        """ GAT-089
+        *Test case for testing creating, listing, deleting bridges*
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/extend_tests/test01_containers_apis.py]: done

**Pull request merged**